src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -152,13 +152,34 @@ public void visitJavadocToken(DetailNode ast) {
         }
         else {
             firstSentence = firstSentence.substring(0, endOfSentence);
+            if (!isPeridAtEndOfSentence(ast)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
+            }
             if (containsForbiddenFragment(firstSentence)) {
                 log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
             }
         }
     }
 
     /**
+     * Finds if period is placed at the end of sentence.
+     * @param ast Javadoc root node.
+     * @return true if period is at the end of sentence or false.
+     */
+    private boolean isPeridAtEndOfSentence(DetailNode ast) {
```

Comment:
@sagar-shah94 
The logic in this method is not correct.
Hint: use JavadocUtils.findFirstToken